EC_finetune/__main__.py

```diff
@@ -428,12 +458,31 @@ def main():
             language_model=language_model,
             orig_model=orig_model
         )
-        training_set = XLImageIdentificationDataset(
-            train_images, args.num_distractors_train, args, tokenizer
-        )
-        valid_set = XLImageIdentificationDataset(
-            valid_images, args.num_distractors_valid, args, tokenizer
-        )
+        if getattr(args, 'ec_input_text', False):
+            training_set = TextInputECDataset(
+                train_images,
+                train_captions,
+                args.num_distractors_train,
+                tokenizer,
+                args,
+                max_length=128
+            )
+            valid_set = TextInputECDataset(
+                valid_images,
+                valid_captions,
+                args.num_distractors_valid,
+                tokenizer,
+                args,
+                max_length=128,
```

Comment:
These two `max_length` shouldn't be hard-coded, but a default arg elsewhere

Reply:
Done!